Enemy progression

felipebrunetti94 opened the pull request:
Changed the order of the index.html, to:
```
<script src="js/game.js"></script>
<script src="js/monster.js"></script>
<script src="js/map.js"></script>
```
And created the `chooseMonsterType()` in the map.js, where level 3 will only spawn normal monsters, level higher than 4 normal and anti Monsters, and higher then 7 few normal, few anti e a lot of plus monsters.

felipebrunetti94 committed:
add monster progression

felipebrunetti94 committed:
fix enemy progression

negreirosleo commented on js/map.js:
```diff
@@ -54,13 +59,42 @@ function generateMonsters(){
     // let monsterType = shuffle([Goose, Ant, Mushroom, Eater, Toast])[0];
-    let monsterType = shuffle([Rock, Paper, Scissors ,Rock_Plus, Paper_Plus, Scissors_Plus, Rock_Anti, Paper_Anti, Scissors_Anti])[0];
+
+     let monsterType = shuffle(chooseMonsterType(level))[0];
```
Look out for the identation here :ok_woman:

negreirosleo replied:
And btw, why `let` and not `const` here?

felipebrunetti94 replied:
yeah, good call. I will change that.

negreirosleo commented on js/map.js:
```diff
@@ -54,13 +59,42 @@ function generateMonsters(){
     let numMonsters = level+1;
     for(let i=0;i<numMonsters;i++){
-        spawnMonster();
+        spawnMonster(level);
```
How about changing that for something like that:
```js
function generateMonsters(level){
  const numMonsters = level+1;
  const monsters = numMonsters.map(() =>
    spawnMonster(level)
  )
}

felipebrunetti94 replied:
The owner of the project, wasn't using map... but i think i will give it a try. Thanks for the review. :D

abnersajr commented on js/map.js:
```diff
@@ -54,13 +59,42 @@ function generateMonsters(){
+  }
+}
+
+function normalMonster(){
```
You can make one function
`shuffleMonster(arg)` arg can ben `[NORMAL,PLUS,ANTI]`
Instead of having 3 different functions.

felipebrunetti94 replied:
Like, unifying `function normalMonster()` , `function plusMonster()` and  `function antiMonster()` ?

negreirosleo replied:
@abnersajr  what would be the difference between calling `shuffleMonster(arg)` and `shuffle(monster_kind)`  ?

felipebrunetti94 replied:
huuumm.... there is a extra layer of randomness I guess, but i think it would be better with fewer functions... I'll give it a try, thanks.

abnersajr replied:
@negreirosleo shuffle can be used to any kind of array. The suffix monster scopes this `shuffle` to a scope of monsters. It's a suggestion based on this new code. Not a mandatory change.
And you can add to the scope of `shuffleMonsters` some kind of validation, default values and so on.
Like I've said this isn't a mandatory change. Just sharing another POV.

felipebrunetti94 committed:
closes #3

nuzcraft commented:
hey @felipebrunetti94, it looks like there are still a few changes to be made based on the feedback from @abnersajr and @negreirosleo (thanks btw!!), so I'll wait to merge until those changes are complete. Everything they've said seems reasonable to me.

nuzcraft commented on js/map.js:
```diff
@@ -54,13 +59,42 @@ function generateMonsters(){
+  if(level < 4){    
+    return normalMonster();
+  }
+  else if(level < 7){
```
at the moment, I believe the game is programmed to reset at level 6. I'll make a new issue to bump it up to 10.

felipebrunetti94 replied:
Sounds good.

felipebrunetti94 committed:
fix(map.js) refact shuffleMonsters

- unify normalMonster(), antiMonster() and plusMonster()

fixed review:
https://github.com/nuzcraft/row-jam-bow/pull/7#issuecomment-545721618

nuzcraft reviewed (approved):
I approve of these changes.

nuzcraft committed:
Merge branch 'master' into enemy-progression
